pkg/rcmanager/subnet.go

```diff
@@ -0,0 +1,300 @@
+				return err
+			})
+			if err != nil {
+				klog.Warningf("Can't update remote subnet in local cluster. err=%v. name=%v", err, v.Name)
+				continue
+			}
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		for _, v := range remove {
+			_ = m.localClusterRamaClient.NetworkingV1().RemoteSubnets().Delete(context.TODO(), v.Name, metav1.DeleteOptions{})
+			if err != nil && !k8serror.IsNotFound(err) {
+				klog.Warningf("Can't delete remote subnet in local cluster. remote subnet name=%v", v.Name)
+			}
+		}
+	}()
+	wg.Wait()
+	return nil
+}
+
+func (m *Manager) RunSubnetWorker() {
+	for m.processNextSubnet() {
+	}
+}
+
+// validate only in local cluster
+func (m *Manager) checkLocalClusterOverlap(subnet *networkingv1.Subnet, subnets []*networkingv1.Subnet) error {
+	for _, s := range subnets {
+		if utils.Intersect(subnet.Spec.Range.CIDR, subnet.Spec.Range.Version, s.Spec.Range.CIDR, s.Spec.Range.Version) {
+			err := errors.Newf("Two subnet intersect. One is from cluster %v, cidr=%v. Another is from lcoal cluster, cidr=%v",
+				m.ClusterName, subnet.Spec.Range.CIDR, s.Spec.Range.CIDR)
+			klog.Error(err.Error())
+			m.recorder.Event(subnet, v1.EventTypeWarning, ReasonSubnetConflict, err.Error())
+			return errors.Newf("Overlap network. overlap with other local cluster subnet")
+		}
+	}
+	return nil
+}
+
+// validate whether the subnet to be added is conflict with remoteSubnet in localCluster
+// validate between all connected domain expect local cluster
+func (m *Manager) checkRemoteSubnetOverlap(subnet *networkingv1.Subnet, rcSubnets []*networkingv1.RemoteSubnet) error {
+	for _, rc := range rcSubnets {
+		if rc.Name == utils.GenRemoteSubnetName(m.ClusterName, subnet.Name) {
+			continue
+		}
+		if utils.Intersect(rc.Spec.Range.CIDR, rc.Spec.Range.Version, subnet.Spec.Range.CIDR, subnet.Spec.Range.Version) {
+			err := errors.Newf("Two subnet intersect. One is from cluster %v, cidr=%v. Another is from cluster %v, cidr=%v",
+				m.ClusterName, subnet.Spec.Range.CIDR, rc.Spec.ClusterName, rc.Spec.Range.CIDR)
+			klog.Error(err.Error())
+			m.recorder.Event(subnet, v1.EventTypeWarning, ReasonSubnetConflict, err.Error())
+			return errors.Newf("Overlap network. overlap with other remoteSubnet")
+		}
+	}
+	return nil
+}
+
+// Reconcile local cluster *networkingv1.RemoteSubnet based on remote cluster's subnet.
+func (m *Manager) diffSubnetAndRCSubnet(subnets []*networkingv1.Subnet, rcSubnets []*networkingv1.RemoteSubnet,
+	networkMap map[string]*networkingv1.Network) (
+	add []*networkingv1.Subnet, update []*networkingv1.RemoteSubnet, remove []*networkingv1.RemoteSubnet) {
+	subnetMap := func() map[string]*networkingv1.Subnet {
+		subnetMap := make(map[string]*networkingv1.Subnet)
+		for _, s := range subnets {
+			subnetMap[s.Name] = s
+		}
+		return subnetMap
+	}()
+
+	for _, v := range rcSubnets {
+		if v.ClusterName != m.ClusterName {
+			continue
+		}
+		if subnet, exists := subnetMap[v.Name]; !exists {
```

Comment:
Yes, Remote Cluster's name can't be empty string and remote subnet's name format is "%v.%v"

Reply:
I mean "_, exists := subnetMap[v.Name]" here will always return false, because the keys of `subnetMap` are `networkingv1.Subnet.Name` rather than `utils.GenRemoteSubnetName(m.ClusterName, subnet.Name)`.